File: BaekJoon/5555_반지.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine();
        int n = sc.nextInt(); sc.nextLine();
        int result = 0;
        for (int i = 0; i < n; i++) {
            String ring = sc.nextLine();
            for (int j = 0; j < ring.length(); j++) {
                if (str.charAt(0) == ring.charAt(j)) {
                    int check = 0; int index = 0;
                    for (int k = j; k < j+str.length(); k++) {
                        if (k == ring.length()) {
                            k = 0;
                        }
                        if (ring.charAt(k) != str.charAt(index)) {
                            check = 1;
                            break;
                        }
                        index++;
                        if (index >= str.length()) break;
                    }
                    if (check == 0) {
                        result++;
                        break;
                    }
                }
            }
        }
        System.out.print(result);
    }
}

File: BaekJoon/5555_반지.kt
import java.util.*
fun main() {
        val sc = Scanner(System.`in`)
        val str = sc.nextLine()
        val n = sc.nextInt()
        sc.nextLine()
        var result = 0
        for (i in 0 until n) {
            val ring = sc.nextLine()
            for (j in 0 until ring.length) {
                if (str[0] == ring[j]) {
                    var check = 0
                    var index = 0
                    var k = j
                    while (k < j + str.length) {
                        if (k == ring.length) {
                            k = 0
                        }
                        if (ring[k] != str[index]) {
                            check = 1
                            break
                        }
                        index++
                        if (index >= str.length) break
                        k++
                    }
                    if (check == 0) {
                        result++
                        break
                    }
                }
            }
        }
        print(result)
    }
